### ISSUE TITLE
Servers added with type "ping" do not go red if they've never responded

### ISSUE THREAD

**Issue by davehamer** [phpservermon/phpservermon#456]
Using the new 'ping' feature in anger today - I've noticed that checks that have NEVER responded to ping, do not go red after 4 failed attempts (my setting)
http://gosub.me.uk/w2gw5 - these have been running for Circa 30 minutes now and are still green despite no connection.
On the other hand - I added a server with type "service" and even though it could not connect at all, it still went red after 4 failed attempts (my setting)

Note: did read some of the other issues - I am running cron every minute! However due to the discrepancy between 'ping' and 'service' types - I believe this may be a bug.

Let me know if you need any more details!

**Comment by baur** [comment]
In my case new 'ping'  feature  not works at all

**Comment by davehamer** [comment]
@baur - You don't get any responses or it doesn't go red on failure?
http://gosub.me.uk/duclb - This server went offline 9 hours ago - no notification in the log, still showing as Green.

**Comment by davehamer** [comment]
I think I've tracked it down, but I have no idea how you guys want to fix it-
https://github.com/phpservermon/phpservermon/blob/develop/src/psm/Util/Server/Updater/StatusUpdater.php#L160
In this line, you specify that rtime = the time taken for a ping response
https://github.com/phpservermon/phpservermon/blob/develop/src/psm/Util/Server/Updater/StatusUpdater.php#L163
In this line, no rtime is defined, thus leaving it as a 'null' value

When passing back to update the database:
https://github.com/phpservermon/phpservermon/blob/develop/src/psm/Util/Server/Updater/StatusUpdater.php#L134
You pass rtime as a null value which results in a mySQL error:
#1265 - Data truncated for column 'rtime' at row 1

I've done a temporary fix by inserting a line 
`$this->rtime = 10
`
At line 160, based on the timeout of 10 seconds that you've set previously for a ping. You may wish to fix this in a more elegant way though.

**Comment by seanvree** [comment]
@BristolITDave  - hey buddy, I tried this workaround, but didn't work for me, but I think I'm confused when you say 

```
I've done a temporary fix by inserting a line
$this->rtime = 10
At line 160, 
```

Can you post the entire file somewhere?

THX!

**Comment by davehamer** [comment]
Hi Sean,
Sorry my line 160 seems to be a bit different - let me give you the before and after code:
Before:
`socket_send($socket, $package, strLen($package), 0);
		if (socket_read($socket, 255)) {
			$this->rtime =  microtime(true) - $starttime;
			$status = true;
		} else {
			$status = false;
		}
		socket_close($socket);`

After:
`socket_send($socket, $package, strLen($package), 0);
		if (socket_read($socket, 255)) {
			$this->rtime =  (microtime(true) - $starttime);
			$status = true;
		} else {
			$this->rtime = 10;
			$status = false;
			$this->error = "Ping timeout";
		}
		socket_close($socket);`

That hasn't pasted well into the comment - so I've also uploaded the entire file to: https://pastebin.com/E1Le8WzQ

**Comment by seanvree** [comment]
Awesome! Thank you, seems like a pretty easy fix right?  Wonder why it hasn't been implemented yet?

**Comment by TimZ99** [comment]
@seanvree no one made a pr I suppose.

I believe it's fixed at #516. Please reopen if it isn't.
